core-java-modules/core-java-8-datetime-3/src/test/java/com/baeldung/datetimecombination/DateTimeCombinationUnitTest.java

```diff
@@ -0,0 +1,38 @@
+package com.baeldung.datetimecombination;
+
+import org.junit.jupiter.api.Test;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.temporal.ChronoField;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class DateTimeCombinationUnitTest {
+    LocalDate date = LocalDate.of(2023, 9, 12);
+    LocalTime time = LocalTime.of(14, 30);
+    LocalDateTime expectedDateTime = LocalDateTime.parse("2023-09-12T14:30");
+
+    @Test
+    public void givenDateAndTime_whenUsingLocalDateTimeOf_thenCombined() {
+        LocalDateTime combinedDateTime = LocalDateTime.of(date, time);
+
+        assertEquals(expectedDateTime, combinedDateTime);
+    }
+
+    @Test
+    public void givenDateAndTime_whenUsingAtTime_thenCombined() {
+        LocalDateTime combinedDateTime = date.atTime(time);
+
+        assertEquals(expectedDateTime, combinedDateTime);
+    }
+
+    @Test
+    public void givenDateAndTime_whenAdjustingTime_thenCombined() {
+        LocalTime adjustedTime = LocalTime.now().with(ChronoField.MINUTE_OF_HOUR, 0);
+        LocalDateTime expectedAdjustedDateTime = LocalDateTime.of(date, adjustedTime);

```

Comment:
By using this, you're essentially only showing that the two previous approaches yield the same result. It would be better to simply define expectedAdjustedDateTime explicitly using LocalDateTime.parse.